### ISSUE TITLE
Urgent Need to understand the Data extracted from Reps API and the formatting 

### ISSUE THREAD

**Issue by mermi** [mozilla/participation-metrics-org#74]
So I have been working on the Community Analytics these days and from the Community Analytics Stories Henrik have sent to me I have been focusing more in Mozilla Activate program, I made this [dashboard](https://analytics.mozilla.community:443/goto/736d77554b2df6bddfc2f3ee1d89b3e7) which contains some interesting charts. My remark is that the Data is not so clear I feel is making me in a confusion. I was thinking to make a chart for the users who organized an event and about Rust for example so making a Filter with Organized an event as activity and Split the chart with Rust activity but this result is the same as you are making a chart for all the users who have been in or making Rust Event, so from that I found myself in a confusion. 
I would love to know more about Data, I think is better is we focus on the data we are extracting first of all and document it, from the data we will have we can know what you can do and the insight we can extract, also we will be able to make some beautiful chart for the people who they are making Stories in Community Analytics Stories. 
I would love to hear from you and what do you think, if anyone is facing the same problem or I am the only one. :)

**Comment by hmitsch** [comment]
Hi @jsmanrique and @sanacl,

can you please have a look at the issue @mermi describes and see if you can answer her points?

Best regards,
   Henrik

**Comment by jgbarah** [comment]
Hi, @mermi, just to clarify: when you say "Data", what do you mean? The "remo2-activities" index, maybe?

**Comment by mermi** [comment]
Yes when I say Data is the fields in "remo2-activities" index @jgbarah

**Comment by jgbarah** [comment]
@mermi, thanks for the clarification. So, you want to use the "remo2-activities" index, for example for items labeled as "Organized an event". But from that point on, I'm not sure what you intend to do. Maybe finding out those items where "Rust" is mentioned? Something else?

I'm sorry about my lack of context, but maybe if we can define as much precisely as possible one scenario, it's easier to find a way. Then we can make it more general.

**Comment by mermi** [comment]
ok, let's say we need to visualize this Scenario: I want to see the Number of people hosting Rust events as part of the MozActivate Campaign=> that means they are Organizing Rust Event, whenwe look to the "remo2-activities" index we have just the activity field and user to work with. if we want to look deeply, looking in the activity field content, you will see you have "Dive Into Rust"[1] and "Organized an Event"[2] so even if you make two filters in the chart with both [1] and [2] is not the right result because organized and Event is general could be related to Rust or another Topic and Dive into Rust is general too could be related to people participated or organized this event. I don't know if the confusion is clear with this Scenario? That's why I said we need to learn more about the Data we extracted from the Reps API to have the right insight and charts.

**Comment by mermi** [comment]
For Example here I found we can Filter Too with MozActivate - Dive Into Rust That means the User made a Report or an Event in the Reps Portal with the Activity "What Do you Do?" == "MozActivate - Dive Into Rust" https://analytics.mozilla.community:443/goto/11efc4d3707dbcfaf1f297d96ea57585 and here we reply to this [Scenario](https://github.com/mozilla/participation-metrics-org/issues/26) but here we first of all we haven't the initiative index in the remo2-activities first of all, also how to be sure that user host the Event this information Could be extracted from a Report or an Event created in the Reps Portal, so here I think we will make like *wrong* decision, I am just confused with that, because as I know we need to learn the Data deeply and what Data can offer and if the Data we have will reply to some story before going to the Analyze part. 
PS: There is any documentation somewhere about the Data you have extracted and each index what are the field they have and a little bit of explanation?  if no I am up to help in that because I think it's so useful for the persons who will analyze the behavior of their program or project in Mozilla.

**Comment by jgbarah** [comment]
Thanks for the description. The problem I see with that scenario, is that in the "remo2-activities" index we have information about the role (`Organizing an event`, for example), in field `activity`, but we don't have categories in it to filter. That's discussed in #66 and that's in the pipeline for being implemented.

So, for now the only thing that can be done is, as you suggest, search for "Rust" (well, in fact `*rust*`) in all the fields. I know that's not a good solution, but if we forget for a moment about that, the fact is that two filters, one on `activity: "Organizing an event`, and another one a general filter on `*rust*`, you get the "AND" of both of them. That is, all items where the person is organizing, and Rust is mentioned.

See for example [this Search](https://analytics.mozilla.community:443/goto/66ead411d8e1372f6948291d82f0c094), which has exactly those two filters (and shows 50 items during the last two years). See the screenshot below

![selection_100](https://cloud.githubusercontent.com/assets/1039693/22161409/7ae29668-df4a-11e6-90c2-b0c0dc08fb0c.png)

But I'm not sure if this is the scenario you are referring to...

**Comment by jgbarah** [comment]
> PS: There is any documentation somewhere about the Data you have extracted and each index what are the field they have and a little bit of explanation? if no I am up to help in that because I think it's so useful for the persons who will analyze the behavior of their program or project in Mozilla.

We get (I guess) all the data available from the Remo API. Then, for now we basically produce one index per API entry point. For example, "remo2-activities" is obtained from the `activities`entry point, where we retrieve JSON documents such as [the one for actiivty 40822](https://reps.mozilla.org/api/remo/v1/activities/40822/).

Currently there is no documentation besides the source code for the [Perceval Remo backend](https://github.com/grimoirelab/perceval-mozilla/blob/master/perceval/backends/mozilla/remo.py) (which is not a documentation after all) and the documentation for the Remo API itself.

I would be more than happy of working with you on writing a good documentation for the information in the index. We have a plan for doing that for all the indexes, and I don't see a reason why we cannot start with this one.

**Comment by mermi** [comment]
So if I have understood correctly how it's working you are extracting the activities from the User page with some information for the user? right?

This one https://analytics.mozilla.community:443/goto/6218eefd6d1da6d37c1020008ea40a2c replies on the Scenario I was talking about but with *rust* filter I don't if it's right or not and also it's really giving the event organized by the "X" user and it's about *rust*!?

Yes, we can start with this index if you want and here you can find if we will need some transformation for some items in each index or not this will help more in the filter phase when you are analyzing.

**Comment by jgbarah** [comment]
> So if I have understood correctly how it's working you are extracting the activities from the User page with some information for the user? right?

Not exactly. We get the data from the Remo API, in this case the [activities API](https://reps.mozilla.org/api/remo/v1/activities/). We get that list, and go through all the items in it, each of them representing an activity, such as [this one](https://reps.mozilla.org/api/remo/v1/activities/37813/).

If you look at the second link, you can see there is some information about the user, and then basically the same fields you can find in the `remo2-activities` index.

**Comment by jgbarah** [comment]
> This one https://analytics.mozilla.community:443/goto/6218eefd6d1da6d37c1020008ea40a2c replies on the Scenario I was talking about but with rust filter I don't if it's right or not and also it's really giving the event organized by the "X" user and it's about rust!?

I'm sorry, but I don't fully understand this. Just to try to clarify: each element in the index is one element we get from the activities API. If I understand correctly that API, each item corresponds to an activity, as recorded by a Remo user. So, if you filter by `activity: Organized an event` and `*rust*` you get all the recorded activities where the user was "organizer", and Rust was mentioned.

If two users record the same activity, with different roles (eg, one organizing, another one attending), we're counting the activity twice if we just count the number of results. If you want to count activities "the right way", instead of just using "count" you should use "unique count" by some field which is unique for a single activity. This is maybe the event url, which I see in the index. But if this is what you are looking for, I guess we need an expert on the Remo API, to make sure this is the case.

**Comment by mermi** [comment]
Yes, I agree that we need an expert on Remo API, because using the solutions we were discussing on the filter phase will bring some different results, I usually do the unique count on activity, and make two filters one for the bar and the other for the chart [column] to get the people who organized *rust* event or people who attended *rust* event. but the confusion show up clearly in the analyze part, you can see here the same filter one with *rust* and the other MozActivate - Dive Into Rust but different result in the chart: 
https://analytics.mozilla.community:443/goto/be2356975cb6a26273da303af27b55a4
and 
https://analytics.mozilla.community:443/goto/67a0100c14f618d912ef92c8dd477dfa

In the same case two or three of them is attending the same number of activity Dive Into Rust: https://analytics.mozilla.community:443/goto/18670f218c8de066836ba300828818f5

that's why I said filtering in analytic part is not giving the right result and we should learn more about the data came from Remo API.

**Comment by MichaelKohler** [comment]
I can see what you mean. I've tried to analyze this for one Rep (https://reps.mozilla.org/reports/rep/Mte90/), he's not matching anything with the *rust* filter, but has one entry for the "Dive Into Rust" initiative. Due to this, I wanted to see which report is in the "Dive Into Rust" activity, without having the keyword "rust" in it. However, on his reports (where the activities are coming from) I can't find any entry in the past 6 months that would say "Dive Into Rust" as an activity. Neither on the according event pages that are linked from the activities. Therefore it should not show one entry for Daniele as far as I can see. This is weird :/

Could somebody with better data skills find out which entry for the "Dive Into Rust" initiative for Daniele Scasciafratte there is in the index? Maybe I'm just blind in the morning ;)

For the descrepancy between "MozActivate - Dive Into Rust" and "Dive Into Rust" I will dive into this further tonight. I thought our change in renaming this should not have had any impact, but this doesn't seem like it. I will see how that record gets saved.

**Comment by jgbarah** [comment]
@MichaelKohler said:

> Could somebody with better data skills find out which entry for the "Dive Into Rust" initiative for Daniele Scasciafratte there is in the index? Maybe I'm just blind in the morning ;)

Maybe the comments in #75 are enough to understand what's happening, and how to do what you need?

**Comment by MichaelKohler** [comment]
Yes, this is okay for me. Maybe @mermi has some more questions.

**Comment by hmitsch** [comment]
Note from a call with Bitergia (which we currently have):
> **Contribution area** (in event report) **= functional area** (in activity) **= Category** (in event)

**Comment by hmitsch** [comment]
Lets settle on **functional area** for this term. Please change all panels accordingly.
